crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -145,4 +147,30 @@ pub mod common_conditions {
     ) -> impl ReadOnlySystem<In = (), Out = bool> {
         condition.pipe(|In(val): In<bool>| !val)
     }
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if the given component type was added to any entities.
+    ///
+    /// Run conditions are evaluated on the main thread, blocking any other systems from running.
+    /// This run condition is relatively expensive, as it iterates over every entity with this component.
+    /// As a result, you likely only want to use this run condition when the number of entities with the component `T` is small.
+    pub fn any_component_added<T: Component>() -> impl FnMut(Query<(), Added<T>>) -> bool {
+        move |query: Query<(), Added<T>>| !query.is_empty()
+    }
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if the value of the given component type has been changed on any entities.
+    ///
+    /// Run conditions are evaluated on the main thread, blocking any other systems from running.
+    /// This run condition is relatively expensive, as it iterates over every entity with this component.
+    /// As a result, you likely only want to use this run condition when the number of entities with the component `T` is small.
+    pub fn any_component_changed<T: Component>() -> impl FnMut(Query<(), Changed<T>>) -> bool {
+        move |query: Query<(), Changed<T>>| !query.is_empty()
+    }
+
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if there are any entities with the given component type removed.
+    pub fn any_component_removed<T: Component>() -> impl FnMut(RemovedComponents<T>) -> bool {
+        move |removals: RemovedComponents<T>| !removals.is_empty()
```

Comment:
Wait. My suggestion was wrong. I did not realize that `RemoveComponents` is now based on events, and therefore the same caveats as my `on_event` run condition apply. The events need to be drained, just checking `is_empty` is not enough. Your old iteration code was correct, and my suggestion is wrong. Please revert it. I'm sorry for misleading you.

Reply:
No problem, reverted!

Reply:
You could also add a comment pointing out that `RemovedComponents` is based on events and therefore events need to be drained (similar to the comment i added for `on_event` in #7579), to avoid confusion and make sure that nobody accidentally messes this up in the future.

Reply:
Agree, added!